### DISCUSSION TITLE
Issue on startup NullRefereneException

### DISCUSSION THREAD

**Discussion by WantClue** [coinfoundry/miningcore#1686] [Q&A] [answered]
I followed the guide to install and build everything. But after I try to run miningcore I get this error and I haven't found a solution about this somewhere 
![image](https://github.com/oliverw/miningcore/assets/86001033/f3f13d43-487c-42ac-9e9f-021bbc94678c)

**Comment by MiningCryptoLive** [comment]
What coin are you setting up?

**Reply by WantClue** [reply]
Monero

**Comment by MiningCryptoLive** [comment]
is the node synced yet? What does your json file look like?

**Reply by WantClue** [reply]
Yes Node is synced and I tried another pool software that's running perfect but only for monero. That's why I want to try this out.
config looks likethis: 
{
    "logging": {
        "level": "info",
        "enableConsoleLog": true,
        "enableConsoleColors": true,
        "logFile": "",
        "logBaseDirectory": "",
        "perPoolLogFile": false
    },
    "banning": {
        "manager": "Integrated",
        "banOnJunkReceive": true,
        "banOnInvalidShares": false
    },
    "persistence": {
        "postgres": {
            "host": "127.0.0.1",
            "port": 5432,
            "user": "miningcore",
            "password": "XXX",
            "database": "miningcore"
        }
    },
    "notifications": {
        "enabled": false,
        "email": {
            "host": "smtp.example.com",
            "port": 587,
            "user": "user",
            "password": "password",
            "fromAddress": "info@yourpool.org",
            "fromName": "support"
        },
        "admin": {
            "enabled": false,
            "emailAddress": "user@example.com",
            "notifyBlockFound": false
        }
    },
    "paymentProcessing": {
        "enabled": true,
        "interval": 600,
        "shareRecoveryFile": "recovered-shares.txt"
    },
    "pools": [{
      "id": "xmr1",
      "enabled": true,
      "coin": "monero",
      "randomXRealm": "xmr1",
      "address": "XXXXX",
      "rewardRecipients": [
        {
          "address": "XXXXXXX",
          "percentage": 1
        }
      ],
      "blockRefreshInterval": 500,
      "clientConnectionTimeout": 600,
      "banning": {
        "enabled": true,
        "time": 600,
        "invalidPercent": 50,
        "checkThreshold": 50
      },
      "ports": {
        "3032": {
          "listenAddress": "0.0.0.0",
          "difficulty": 7500,
          "name": "CPU Mining",
          "varDiff": {
            "minDiff": 1000,
            "maxDiff": null,
            "targetTime": 15,
            "retargetTime": 90,
            "variancePercent": 30
          }
        },
        "3034": {
          "listenAddress": "0.0.0.0",
          "difficulty": 200000,
          "name": "High-End Multi GPU Mining",
          "varDiff": {
            "minDiff": 200000,
            "maxDiff": null,
            "targetTime": 15,
            "retargetTime": 120,
            "variancePercent": 30
          }
        }
      },
      "daemons": [
        {
          "host": "127.0.0.1",
          "port": 28082,
           "user": "",
          "password": ""
       },
        {
          "host": "127.0.0.1",
          "port": 28084,
          "user": "",
          "password": "",
          "category": "wallet"
        }
      ],
      "paymentProcessing": {
        "enabled": true,
        "minimumPayment": 0.25,
        "minimumPaymentToPaymentId": 1.0,
        "payoutScheme": "PPLNS",
        "payoutSchemeConfig": {
          "factor": 2.0
        }
      }
    }]
}

**Reply by MiningCryptoLive** [reply]
Have you looked at the debug log? Looks like something didn't build correctly.

**Reply by WantClue** [reply]
No it was not producing it or I'm unableto find it

**Reply by WantClue** [reply]
I'm running this on ubuntu 22.04 with NET 17 could this be an issue ?

**Comment by MiningCryptoLive** [comment]
you need to use dotnet 6.0

**Comment by WantClue** [comment, marked as answer]
Still not gonna be able to start miningcore

**Reply by WantClue** [reply]
Got it working

**Comment by MiningCryptoLive** [comment]
What was the issue?

On Sun, May 14, 2023 at 6:38 AM WantClue ***@***.***> wrote:

> Got it working
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/oliverw/miningcore/discussions/1686#discussioncomment-5896300>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIAKRXJQGMW3OGO32QIXXDXGCYZFANCNFSM6AAAAAAX5JEAQU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Randy Hasson
199 East Markison Avenue
Columbus, Ohio 43207

**Reply by WantClue** [reply]
I was missing my API settings

**Comment by MiningCryptoLive** [comment]
That’s definitely an important part.
What’s your pool address?

On Sun, May 14, 2023 at 6:48 AM WantClue ***@***.***> wrote:

> I was missing my API settings
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/oliverw/miningcore/discussions/1686#discussioncomment-5896339>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIAKRRY6D2H5FMLTINPWDLXGCZ6NANCNFSM6AAAAAAX5JEAQU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Randy Hasson
199 East Markison Avenue
Columbus, Ohio 43207

**Reply by WantClue** [reply]
Currently working on a couple of things. I send it to you when I'm done
